contracts/InverseVester.sol

```diff
@@ -7,78 +7,101 @@ import "./interfaces/IInv.sol";
 /**
  * @title Inverse token vesting contract
  * @author Inverse Finance
- * @notice Base contract for vesting agreement on INV tokens
- *         Reverse vesting means tokens are owned but the recipient
- *         and can be used for voting.
+ * @notice Contract for vesting agreement on INV tokens
  * @dev    Vesting calculation is linear
  */
 contract InverseVester is Ownable {
     using SafeERC20 for IInv;
 
     uint256 public constant DAY = 1 days;
 
-    // @dev Timestamp for the start of this vesting agreement
+    /// @dev Timestamp for the start of this vesting agreement
     uint256 public vestingBegin;
 
-    // @dev Timestamp for the end of this vesting agreement
+    /// @dev Timestamp for the end of this vesting agreement
     uint256 public vestingEnd;
 
-    // @dev Timestamp for the last time vested tokens were claimed
+    /// @dev Timestamp for the last time vested tokens were claimed
     uint256 public lastClaimTimestamp;
 
-    // @dev Total amount to be vested
+    /// @dev Total amount to be vested
     uint256 public immutable vestingAmount;
 
-    // @dev Inverse finance treasury token
-    IInv public immutable inv;
+    /// @dev By how long the vesting should be delayed after activating this contract
+    ///      This can be used for multi year vesting agreements
+    uint16 public immutable vestingStartDelayInDays;
+
+    /// @dev Inverse finance governance timelock contract
+    address public timelock;
 
-    // @dev Amount of days the vesting period will last
+    /// @dev Amount of days the vesting period will last
     uint16 public immutable vestingDurationInDays;
 
-    // @dev Whether this is a reverse vesting agreement
+    /// @dev Whether this is a reverse vesting agreement
     bool public immutable reverseVesting;
 
+    /// @dev Whether this is vesting agreement can be interrupted
+    bool public immutable interruptible;
+
+    /// @dev Inverse finance treasury token
+    IInv public immutable inv;
+
+    /**
+     * @dev Prevents non timelock from calling a method
+     */
+    modifier onlyTimelock() {
+        require(msg.sender == timelock, "InverseVester:ACCESS_DENIED");
+        _;
+    }
+
     constructor(
-        IInv _inv,
-        uint256 _vestingAmount,
-        uint16 _vestingDurationInDays,
-        bool _reverseVesting
+        address inv_,
+        address timelock_,
+        uint256 vestingAmount_,
```

Comment:
If we're already guarding against accidentally filling in the zero address, we might as well guard against setting a zero value for `vestingAmount`, which would make the entire contract a waste of gas.